### PR TITLE
Added tags list feature

### PR DESCRIPTION
**Pull request by kon3gor**
Added tags list feature to nodes list as requested in #1539.

### Preview
Screenshot 2023-04-04 at 20 24 57
Screenshot 2023-04-04 at 20 25 05

https://user-images.githubusercontent.com/37112953/229870051-cd4f05b3-6399-4407-a3fa-5469932ae5c2.mov

### CODE REVIEW

**Comment by J-Jamet** [comment]
Pretty cool :)

File: app/src/main/java/com/kunzisoft/keepass/view/TagsListView.kt [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    context: Context, attrs: AttributeSet? = null
+) : ConstraintLayout(context, attrs) {
+
+    var textColor: Int? = null
+        set(value) {
+            if (field == value) {
+                return
+            }
+            field = value
+            expandBtn?.setColorFilter(value!!)
```
Comment (kon3gor) [inline app/src/main/java/com/kunzisoft/keepass/view/TagsListView.kt:38]: eh, missed double bang, should be removed

**Comment by J-Jamet** [comment]
Not yet, I have a lot of work at the moment, I put your MR for version 4.1.0, I have to manage the separation in module for version 4.0.0. I know it's long, sorry, but it will be processed, don't worry, thanks a lot for your work.

File: app/src/main/java/com/kunzisoft/keepass/view/ViewUtil.kt [hunk 2]
```diff
@@ -166,6 +166,38 @@ fun View.expand(animate: Boolean = true,
...
+        if (!alreadyVisible && layoutParams.height > 0) {
+            visibility = View.VISIBLE
+            alreadyVisible = true
+        }
+        requestLayout()
+    }
+    AnimatorSet().apply {
+        play(slideAnimator)
+        interpolator = AccelerateDecelerateInterpolator()
+        addListener(object: Animator.AnimatorListener {
```
Comment (J-Jamet) [inline app/src/main/java/com/kunzisoft/keepass/view/ViewUtil.kt:190]: No StubAnimatorListener?

**Comment by J-Jamet** [comment]
I've just done an initial check with a merge to resolve conflicts with the new features in develop version 4.0.0. : https://github.com/Kunzisoft/KeePassDX/tree/kon3gor-feature/tags_in_nodes_list

I think I'll do a simpler implementation than yours, i.e. make the tags scroll horizontally, so there's no need to expand and it'll be easier to use.
For the moment it's only for entries, I'm going to make it compatible with groups too.

You're coding well, but I'm going to simplify. Thanks for your work, I'm starting from your branch, as I'm going to make some big changes your code will be in any case in the project.

**Pull request closed without merging**